contracts/bprotocol/NonUpgradeableProxy.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.6.0;
```

Comment:
this looks too complicated, would be best to use this:
https://github.com/gnosis/safe-contracts/blob/development/contracts/proxies/GnosisSafeProxy.sol